ash/src/device.rs

```diff
@@ -18,11 +18,10 @@ pub struct Device {
 }
 
 impl Device {
-    pub unsafe fn load(instance_fn: &vk::InstanceFnV1_0, device: vk::Device) -> Self {
-        let load_fn = |name: &std::ffi::CStr| {
-            mem::transmute((instance_fn.get_device_proc_addr)(device, name.as_ptr()))
-        };
-
+    pub unsafe fn load_with(
+        load_fn: impl FnMut(&core::ffi::CStr) -> *const c_void + Copy,
```

Comment:
Let's avoid the `Copy` constraint by passing a unique reference down.

Reply:
By changing the signature of all the `fn load()`s? Sure, we can do that, there's unlikely to be any harm in that.

Reply:
Oh, I hadn't even seen that. For that reason it also wouldn't make sense for us to take `&mut impl FnMut`, as it requires the caller to use `&mut |name| { ... }`. In case they end up in a similar situation they can pass an `&mut` of _their_ closure as well.